src/gui/widgets/LinkedModelGroupLayout.cpp

```diff
@@ -0,0 +1,303 @@
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and The Qt Company. For licensing terms
+** and conditions see https://www.qt.io/terms-conditions. For further
+** information use the contact form at https://www.qt.io/contact-us.
+**
+** BSD License Usage
+** Alternatively, you may use this file under the terms of the BSD license
+** as follows:
+**
+** "Redistribution and use in source and binary forms, with or without
+** modification, are permitted provided that the following conditions are
+** met:
+**   * Redistributions of source code must retain the above copyright
+**     notice, this list of conditions and the following disclaimer.
+**   * Redistributions in binary form must reproduce the above copyright
+**     notice, this list of conditions and the following disclaimer in
+**     the documentation and/or other materials provided with the
+**     distribution.
+**   * Neither the name of The Qt Company Ltd nor the names of its
+**     contributors may be used to endorse or promote products derived
+**     from this software without specific prior written permission.
+**
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+** "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+** LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+** A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+** OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+** LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+** DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+** THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+** (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+
+#include "LinkedModelGroupLayout.h"
+
+#include <QWidget>
+#include <QLayoutItem>
+#include <QLineEdit>
+#include <QRect>
+#include <QString>
+
+// TODO: rename: ModelGroupLayout
+
+constexpr const int LinkedModelGroupLayout::m_minWidth;
+
+LinkedModelGroupLayout::LinkedModelGroupLayout(QWidget *parent, int margin, int hSpacing, int vSpacing)
+	: QLayout(parent), m_hSpace(hSpacing), m_vSpace(vSpacing),
+	  m_searchBar(new QLineEdit(parent))
+{
+	setContentsMargins(margin, margin, margin, margin);
+	m_searchBar->setPlaceholderText("filter");
+	connect(m_searchBar, SIGNAL(textChanged(const QString&)),
+		this, SLOT(onTextChanged(const QString& )));
+	addWidget(m_searchBar);
+	m_searchBar->setHidden(true); // nothing to filter yet
+}
+
+LinkedModelGroupLayout::~LinkedModelGroupLayout()
+{
+	QLayoutItem *item;
+	while ((item = takeAt(0))) { delete item; }
```

Comment:
At first glance it may not be clear what this loop condition is. Maybe be explicit that this is looking at item's value.

Reply:
What do you mean by looking? It's just removing all items from the list and deletes them.

Reply:
> 
> 
> What do you mean by looking? It's just removing all items from the list and deletes them.

`while ((item = takeAt(0)))` implicit checking if not 0/false/null.
`while ((item = takeAt(0)) != 0)` explicit check. The extra parentheses can be overlooked on glance. Explicit > implicit.

Reply:
Oh, you mean *that*. Thanks for explaining, I'll fix it.